fearlessTests/Modules/Staking/StakingRewardPayouts/StakingRewardPayoutsTests.swift

```diff
@@ -1,12 +1,170 @@
+                }
+            }
+        }
+
+        stub(viewModelFactory) { stub in
+            when(stub).createPayoutsViewModel(payoutsInfo: any(), priceData: any()).then { _ in
+                LocalizableResource { _ in
+                    StakingPayoutViewModel(
+                        cellViewModels: [],
+                        bottomButtonTitle: ""
+                    )
+                }
+            }
+        }
+
+        let showRewardDetailsExpectation = XCTestExpectation()
+        stub(wireframe) { stub in
+            when(stub)
+                .showRewardDetails(from: any(), payoutInfo: any(), activeEra: any(), historyDepth: any(), chain: any())
+                .then { _ in
+                    showRewardDetailsExpectation.fulfill()
+                }
+        }
+
+        // when
+        let tableView = viewController.rootView.tableView
+        viewController.loadView()
+        tableView.delegate?.tableView?(tableView, didSelectRowAt: IndexPath(row: 0, section: 0))
+
+        // then
+        wait(for: [showRewardDetailsExpectation], timeout: Constants.defaultExpectationDuration)
+    }
+
+    func testShowPayoutConfirmation_whenUserTapsPayoutButton() {
+        // given
+        let interactor = MockStakingRewardPayoutsInteractorInputProtocol()
+        let wireframe = MockStakingRewardPayoutsWireframeProtocol()
+
+        let viewModelFactory = MockStakingPayoutViewModelFactoryProtocol()
+        let presenter = StakingRewardPayoutsPresenter(
+            chain: .westend,
+            viewModelFactory: viewModelFactory
+        )
+        presenter.wireframe = wireframe
+        presenter.interactor = interactor
+
+        let viewController = StakingRewardPayoutsViewController(presenter: presenter, localizationManager: nil)
+        presenter.view = viewController
+
+        stub(interactor) { stub in
+            when(stub).setup().then {
+                if case let Result.success(payoutsInfo) = PayoutRewardsServiceStub.dummy().result {
+                    presenter.didReceive(result: .success(payoutsInfo))
+                }
+            }
+        }
+
+        stub(viewModelFactory) { stub in
+            when(stub).createPayoutsViewModel(payoutsInfo: any(), priceData: any()).then { _ in
+                LocalizableResource { _ in
+                    StakingPayoutViewModel(
+                        cellViewModels: [],
+                        bottomButtonTitle: "Payout all"
+                    )
+                }
+            }
+        }
+
+        let showPayoutConfirmationExpectation = XCTestExpectation()
+        stub(wireframe) { stub in
+            when(stub)
+                .showPayoutConfirmation(for: any(), from: any())
+                .then { _ in
+                    showPayoutConfirmationExpectation.fulfill()
+                }
+        }
+
+        // when
+        let payoutButton = viewController.rootView.payoutButton
```

Comment:
The same is here - we need to use stub of the view and just call a corresponding presenter method